pkg/runtime/runtime.go

```diff
@@ -630,11 +628,12 @@ func (a *DaprRuntime) sendToDeadLetterIfConfigured(name string, msg *pubsub.NewM
 	err = a.Publish(req)
 	if err != nil {
 		log.Errorf("error sending message to dead letter, origin topic: %s dead letter topic %s err: %w", msg.Topic, deadLetterTopic, err)
+		return err
 	}
-	return true, err
+	return nil
```

Comment:
nitpicking:

How about removing the `return err` from line 631? instead `return err` on line 633 directly, the behavior is pretty much the same

Reply:
You're totally rightYou're right that it would be the same. The reason why I don't do that, in general, is because, especially on a project like Dapr with lots of contributors, it can help prevent future regressions :)

Example, following your suggestion:

```go
err = a.Publish(req)
if err != nil {
	log.Errorf("error sending message to dead letter, origin topic: %s dead letter topic %s err: %w", msg.Topic, deadLetterTopic, err)
}
return err
```

Now, if someone wants to add more code before returning, they will need to re-add a `return err` within the block.

```go
err = a.Publish(req)
if err != nil {
	log.Errorf("error sending message to dead letter, origin topic: %s dead letter topic %s err: %w", msg.Topic, deadLetterTopic, err)
	return err
}
// Do something else
return nil
```

I've seen it happen before many times where people forget the `return err`. So as a good practice I try to prevent these and save others (and myself!) from potentially making a mistake in the future.